lib/id_manager.c

```diff
@@ -203,6 +222,34 @@ struct mptcpd_idm *mptcpd_idm_create(void)
                 idm = NULL;
         }
 
+        if (_idm_hash_seed == 0) {
+#ifdef HAVE_GETRANDOM
+                /*
+                  Randomize the hash seed value with data from the
+                  kernel random number generator.
+                */
+                ssize_t const len =
+                        getrandom(&_idm_hash_seed,
+                                  sizeof(_idm_hash_seed),
+                                  GRND_NONBLOCK);
+
+                if (len == -1) {  // unlikely
+                        char errmsg[80];
+                        int const r =
+                                strerror_r(errno,
+                                           errmsg,
+                                           L_ARRAY_SIZE(errmsg));
+
+                        l_error("Unable to generate ID manager "
+                                "hash seed: %s",
+                                r == 0 ? errmsg : "<unknown error>");
+                }
+#else
+                // Fallback on current time as seed (not desirable).
+                _idm_hash_seed = (uint32_t) time(NULL);
```

Comment:
I'd much prefer fallback to `l_getrandom_uint32()`. That function even has additional fallback to `random()` if the syscall fails.

Reply:
I'd rather drop the fallback completely, and stick with `l_getrandom_uint32()`.